v0/destinations/shynet/transform.js

```diff
@@ -0,0 +1,102 @@
+/* eslint-disable no-underscore-dangle */
+const { EventType } = require("../../../constants");
+
+const {
+  constructPayload,
+  defaultRequestConfig,
+  removeUndefinedAndNullValues,
+  getSuccessRespEvents,
+  getErrorRespEvents,
+  CustomError,
+  generateUUID
+} = require("../../util");
+
+const { ConfigCategory, mappingConfig } = require("./config");
+
+// build final response
+function buildResponse(payload, endpoint) {
+  const response = defaultRequestConfig();
+  response.endpoint = endpoint;
+  response.body.JSON = removeUndefinedAndNullValues(payload);
+  return {
+    ...response,
+    headers: {
+      "Content-Type": "application/json"
+    }
+  };
+}
+
+// process page call
+function processPage(message, shynetServiceUrl) {
+  const requestJson = constructPayload(
+    message,
+    mappingConfig[ConfigCategory.PAGE.name]
+  );
+
+  // generating UUID
+  requestJson.idempotency = generateUUID();
```

Comment:
Can't we use `anonymousId` or `userId`?

Reply:
I believe this is a random Id they need for each request

Reply:
Can't we use messageId?